### PR TITLE
add token to the request db header and account id into the response

### PR DESCRIPTION
**Pull request by YiMysty**
Please have a look at our [guidelines for contributions](https://github.com/Microsoft/AppCenter-SDK-Android/blob/develop/CONTRIBUTING.md) and consider the following before you submit the PR:

* [ ] Has `CHANGELOG.md` been updated?
* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
* [x] Did you add unit tests?
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?
* [ ] Did you check UI tests on the sample app? They are not executed on CI.

## Description

Implement token exchange API, using user identity when available.

## Related PRs or issues

[AB#58606](https://msmobilecenter.visualstudio.com/web/wi.aspx?pcguid=27bf89de-9697-418a-accf-1aa125b22914&id=58606)

### CODE REVIEW

File: sdk/appcenter-storage/src/main/java/com/microsoft/appcenter/storage/Storage.java [hunk 1]
```diff
@@ -68,6 +68,8 @@
 
     private HttpClient mHttpClient;
 
+    private Context mConntext;
```
Comment (YiMysty) [inline sdk/appcenter-storage/src/main/java/com/microsoft/appcenter/storage/Storage.java]: do not know why other property name start with `m`... thus add m here. I thought `m` is short for mock?

File: sdk/appcenter-storage/src/main/java/com/microsoft/appcenter/storage/client/TokenExchange.java [hunk 2]
```diff
@@ -65,17 +70,22 @@ public static String buildAppCenterGetDbTokenBodyPayload(String partition) {
     public static synchronized ServiceCall getDbToken(
             final String partition,
             HttpClient httpClient,
+            Context context,
             String apiUrl,
             final String appSecret,
             TokenExchangeServiceCallback serviceCallback) {
         AppCenterLog.debug(LOG_TAG, "Getting a resource token from App Center...");
         String url = apiUrl + GET_TOKEN_PATH_FORMAT;
+        final String userToken = TokenStorageFactory.getTokenStorage(context).getToken();
```
Comment (YiMysty) [inline sdk/appcenter-storage/src/main/java/com/microsoft/appcenter/storage/client/TokenExchange.java]: This context is get when the Storage service initialized, is it using the same storage as identity module?

File: sdk/appcenter-storage/build.gradle [hunk 3]
```diff
@@ -10,6 +10,7 @@ dependencies {
     api project(':sdk:appcenter')
 
     implementation 'com.google.code.gson:gson:2.8.5'
+    compile project(path: ':sdk:appcenter-identity')
```
Comment (guperrot) [inline sdk/appcenter-storage/build.gradle]: Remove that, we must not depend on Identity directly, everything you need is on the core.

File: sdk/appcenter-storage/src/main/java/com/microsoft/appcenter/storage/Storage.java (hunk 1 above)
Comment (guperrot) [reply sdk/appcenter-storage/src/main/java/com/microsoft/appcenter/storage/Storage.java]: m is android convention for member (variable) as opposed to s for static.

Comment (guperrot) [inline sdk/appcenter-storage/src/main/java/com/microsoft/appcenter/storage/Storage.java]: typo: mContext

Comment (jwallra) [inline sdk/appcenter-storage/src/main/java/com/microsoft/appcenter/storage/Storage.java]: Here's the convention: https://source.android.com/setup/contribute/code-style#follow-field-naming-conventions

```suggestion
    private Context mContext;
```

File: sdk/appcenter-storage/src/main/java/com/microsoft/appcenter/storage/client/TokenExchange.java (hunk 2 above)
Comment (guperrot) [reply sdk/appcenter-storage/src/main/java/com/microsoft/appcenter/storage/client/TokenExchange.java]: This pattern is the right one to use.

File: sdk/appcenter-storage/src/main/java/com/microsoft/appcenter/storage/models/TokenResult.java [hunk 4]
```diff
@@ -123,6 +130,17 @@ public TokenResult withExpirationTime(Date expiresOn) {
         return this;
     }
 
+    /**
+     * Set the account id.
+     *
+     * @param accountId Account id value to be set.
+     * @return The TokenResult object itself.
+     */
+    public TokenResult withAccountId(String accountId) {
```
Comment (jwallra) [inline sdk/appcenter-storage/src/main/java/com/microsoft/appcenter/storage/models/TokenResult.java]: The "with" convention is only used in apple SDK.

```suggestion
    public TokenResult setAccountId(String accountId) {
```

File: sdk/appcenter-storage/src/main/java/com/microsoft/appcenter/storage/Storage.java (hunk 1 above)
Comment (guperrot) [reply sdk/appcenter-storage/src/main/java/com/microsoft/appcenter/storage/Storage.java]: We use that convention everywhere except for JSON models.

File: sdk/appcenter-storage/src/main/java/com/microsoft/appcenter/storage/models/TokenResult.java [hunk 5]
```diff
@@ -212,6 +230,14 @@ public String status() {
         return this.status;
     }
 
+    /**
+     * Get account id.
+     * @return The account id value.
+     */
+    public String accountId() {
```
Comment (jwallra) [inline sdk/appcenter-storage/src/main/java/com/microsoft/appcenter/storage/models/TokenResult.java]:
```suggestion
    public String getAccountId() {
```

File: sdk/appcenter-storage/build.gradle (hunk 3 above)
Comment (guperrot) [reply sdk/appcenter-storage/build.gradle]: You're using the wrong class actually.

File: sdk/appcenter-storage/src/main/java/com/microsoft/appcenter/storage/client/TokenExchange.java (hunk 2 above)
Comment (guperrot) [reply sdk/appcenter-storage/src/main/java/com/microsoft/appcenter/storage/client/TokenExchange.java]: You need to use `AuthTokenContext` instead.

File: sdk/appcenter-storage/src/main/java/com/microsoft/appcenter/storage/models/TokenResult.java (hunk 4 above)
Comment (YiMysty) [reply sdk/appcenter-storage/src/main/java/com/microsoft/appcenter/storage/models/TokenResult.java]: sure, will fix it latter :) overall files, currently the naming is in all models

File: sdk/appcenter-storage/src/main/java/com/microsoft/appcenter/storage/models/TokenResult.java (hunk 5 above)
Comment (YiMysty) [reply sdk/appcenter-storage/src/main/java/com/microsoft/appcenter/storage/models/TokenResult.java]: will fix latter in another pr~

File: sdk/appcenter-storage/src/main/java/com/microsoft/appcenter/storage/models/TokenResult.java (hunk 4 above)
Comment (guperrot) [reply sdk/appcenter-storage/src/main/java/com/microsoft/appcenter/storage/models/TokenResult.java]: Let's get that merged, then change everywhere when noone is modifying it.

File: sdk/appcenter-storage/src/main/java/com/microsoft/appcenter/storage/models/TokenResult.java (hunk 5 above)
Comment (guperrot) [reply sdk/appcenter-storage/src/main/java/com/microsoft/appcenter/storage/models/TokenResult.java]: Let's get that merged, then change everywhere when noone is modifying it.

**Pull request merged**